crates/solvers/src/domain/solver/dex/mod.rs

```diff
@@ -86,40 +101,61 @@ impl Dex {
             .filter_map(future::ready)
     }
 
+    async fn try_solve(
+        &self,
+        order: &Order,
+        dex_order: &dex::Order,
+        tokens: &auction::Tokens,
+        gas_price: auction::GasPrice,
+    ) -> Result<dex::Swap, infra::dex::Error> {
+        let slippage = self.slippage.relative(&dex_order.amount(), tokens);
+        self.dex
+            .swap(dex_order, &slippage, tokens, gas_price)
+            .await
+            .map_err(|err| {
+                match &err {
+                    err @ infra::dex::Error::NotFound => {
+                        if order.partially_fillable {
+                            // Only adjust the amount to try next if we are sure the API worked
+                            // correctly yet still wasn't able to provide a
+                            // swap.
+                            self.fills.reduce_next_try(order.uid);
+                        } else {
+                            tracing::debug!(?err, "skipping order");
+                        }
+                    }
+                    err @ infra::dex::Error::OrderNotSupported => {
+                        tracing::debug!(?err, "skipping order")
+                    }
+                    err @ infra::dex::Error::RateLimited => {
+                        tracing::debug!(?err, "encountered rate limit")
```

Comment:
Is this codepath even possible? I believe this error would be caught by the rate limiter component and returned in `solve_order`

Reply:
The error is handled even before the rate limiter and used later in the `requires_back_off` closure by the limiter. The rate limiter either fails fast if the back-off timeout is still not over or executes the task and returns the RateLimited error.